DesignPatternExamples/Assets/Design Patterns/Builder/Scripts/IAirshipBuilder.cs

```diff
@@ -0,0 +1,15 @@
+using System.Collections.Generic;
+
+namespace Builder
+{
+    public interface IAirshipBuilder
+    {
+        List<IAirshipModule> airshipModules { get; set; }
```

Comment:
why do you need a setter for this list?

Consider using `IList<IAirshipModule>`, and in the implementation uses a backingField which implements `List<IAirshipModule>`.


Consider doing the following in the implementation:
```
        private List<IAirshipModule> _airshipModules = new List<IAirshipModule>();
        public IList<IAirshipModule> airshipModules => _airshipModules.AsReadOnly();
```

The reasoning is that even without the Setter, lists can be manipulated by using the `List` class methods (e.g. `Add`). By returning an IList, and returning the `ReadOnly` version you prevent that from happening.

Reply:
Incredible! Thanks!